src/baseline_bots/bots/smart_order_accepter_bot.py

```diff
@@ -325,16 +326,69 @@ async def respond_to_alliance_messages(self, messages_data: MessagesData) -> Non
         :param messages_data: Message Data object to add messages
         """
         unique_senders = {}
-        for sender_message_tuple in self.alliances.values():
+        for sender_message_tuple in self.alliances_prps.values():
             for sender, message in sender_message_tuple:
                 unique_senders[sender] = message
+
+        power_stance = self.stance.stance[self.power_name]
+        for sender, message in unique_senders.items():
+            if sender == self.power_name:
+                continue
+
+            # if the stance value is lower than 5x enemy_threshold (-2.5 by default)
+            # we will reject the alliance proposal
+            if power_stance[sender] <= 5 * self.enemy_threshold:
+                await self.send_message(sender, str(REJ(message)), messages_data)
+                print(f"Alliance rejected: {sender}")
+            # otherwise accept the alliance proposal and update stance to 2.5 * ally_threshold (+2.5 by default)
+            else:
+                self.stance.update_stance(
+                    self.power_name, sender, 2.5 * self.ally_threshold
+                )
+                await self.send_intent_log(
+                    "My stance to {} becomes {} after accepting the alliance proposal.".format(
+                        sender, 2.5 * self.ally_threshold
+                    )
+                )
+                await self.send_message(sender, str(YES(message)), messages_data)
+                print(f"Alliance accepted: {sender}")
```

Comment:
Instead of printing this, add this to the intent logger. I'm making a change to send all intent logging to console for easier debugging.